### PR TITLE
ConfirmationRequired flag on MGMT API

### PR DESCRIPTION
**Pull request by herdt-michael**
Introduce confirmationRequired flag on Rollouts/TargetFilter/Distribution/Target API.

Signed-off-by: Michael Herdt <Michael.Herdt@bosch.io>

### CODE REVIEW

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutManagement.java [hunk 1]
```diff
@@ -251,6 +252,11 @@ private Rollout createRolloutGroups(final int amountOfGroups, final RolloutGroup
         return rolloutRepository.save(savedRollout);
     }
 
+    private boolean isUserConsentEnabled() {
```
Comment (stefbehl) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutManagement.java:255]: Wondering about the name (which reflects one use case). would have expected sth like "isConfirmationSupportEnabled" or "isAssignmentConfirmationsEnabled"...

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetFilterQueryManagement.java [hunk 2]
```diff
@@ -279,10 +280,16 @@ public TargetFilterQuery updateAutoAssignDS(final AutoAssignDistributionSetUpdat
...
             targetFilterQuery.setAutoAssignWeight(update.getWeight());
-            targetFilterQuery.setConfirmationRequired(update.getConfirmationRequired());
+            final boolean confirmationRequired = update.isConfirmationRequired() == null ? isUserConsentEnabled()
+                    : update.isConfirmationRequired();
+            targetFilterQuery.setConfirmationRequired(confirmationRequired);
         }
         return targetFilterQueryRepository.save(targetFilterQuery);
     }
+    
+    private boolean isUserConsentEnabled(){
```
Comment (stefbehl) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetFilterQueryManagement.java:290]: Wondering about the name (which reflects one use case). would have expected sth like "isConfirmationSupportEnabled" or "isAssignmentConfirmationsEnabled"...

**Review by stefbehl — APPROVED** [review]
Looks good. Thanks @herdt-michael !

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetFilterQueryManagement.java (hunk 2 above)
Comment (herdt-michael) [reply hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetFilterQueryManagement.java:290]: Will be discussed after implementing all features of the user consent story

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutManagement.java (hunk 1 above)
Comment (herdt-michael) [reply hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutManagement.java:255]: Will be discussed after implementing all features of the user consent story

**Pull request merged**